src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs

```diff
@@ -1515,38 +1510,77 @@ private void ProcessErrorThread(object state)
-                            PSRemotingErrorId.IPCServerProcessReportedError,
-                            RemotingErrorIdStrings.IPCServerProcessReportedError,
-                            string.IsNullOrEmpty(error) ?
-                                RemotingErrorIdStrings.SSHClientEndNoErrorMessage
-                                : StringUtil.Format(RemotingErrorIdStrings.SSHClientEndWithErrorMessage, error));
-                        RaiseErrorHandler(new TransportErrorOccuredEventArgs(psrte, TransportMethodEnum.CloseShellOperationEx));
-                        CloseConnection();
-                    }
+
+                    // Any SSH client error results in a broken session.
+                    PSRemotingTransportException psrte = new PSRemotingTransportException(
+                        PSRemotingErrorId.IPCServerProcessReportedError,
+                        RemotingErrorIdStrings.IPCServerProcessReportedError,
+                        StringUtil.Format(RemotingErrorIdStrings.SSHClientEndWithErrorMessage, error));
+                    HandleSSHError(psrte);
                 }
             }
-            catch (ObjectDisposedException) { }
             catch (Exception e)
             {
                 string errorMsg = (e.Message != null) ? e.Message : string.Empty;
                 _tracer.WriteMessage("SSHClientSessionTransportManager", "ProcessErrorThread", Guid.Empty,
                     "Transport manager error thread ended with error: {0}", errorMsg);
+
+                PSRemotingTransportException psrte = new PSRemotingTransportException(
+                    StringUtil.Format(RemotingErrorIdStrings.SSHClientEndWithErrorMessage, errorMsg), 
+                    e);
+                HandleSSHError(psrte);
+            }
+        }
+
+        private void HandleSSHError(PSRemotingTransportException psrte)
+        {
+            RaiseErrorHandler(new TransportErrorOccuredEventArgs(psrte, TransportMethodEnum.CloseShellOperationEx));
+            CloseConnection();
+        }
+
+        private static string ReadError(StreamReader reader)
+        {
+            // Blocking read from StdError stream
+            string error = reader.ReadLine();
+
+            if (string.IsNullOrEmpty(error) ||
+                error.IndexOf("WARNING:", StringComparison.OrdinalIgnoreCase) > -1)
+            {
+                // Handle as interactive warning message
+                Console.WriteLine(error);
+                return string.Empty;
+            }
+
+            // SSH may return a multi-line error message
+            System.Text.StringBuilder sb = new Text.StringBuilder(error);
+            var running = true;
+            while (running)
+            {
+                try
+                {
+                    var task = reader.ReadLineAsync();
+                    if (task.Wait(1000) && (task.Result != null))
+                    {
+                        sb.Append("\r\n");
```

Comment:
Definitely.  I'll make the change.  Also I might be able to use ReadAsync instead of ReadLineAsync() in which case existing new lines are preserved.  This just affects how the error message appears in the exception so it is not too critical.